### PR TITLE
ruby: update to 3.1.3

### PR DESCRIPTION
**Pull request by bradfordzhang**
Add a comment about confusing code
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

Signed-off-by: Bradford Zhang <zyc@zyc.name>

### CODE REVIEW

File: lang/ruby/patches/001-fix-build-with-libressl-3.5.patch [hunk 1]
```diff
@@ -1,6 +1,6 @@
-From e25fb0d0d86da5a9398ebdc9216b2ea89f80fa3d Mon Sep 17 00:00:00 2001
-From: Jeremy Evans <code@jeremyevans.net>
-Date: Fri, 25 Mar 2022 13:11:31 -0700
+From b1641baee099805a07c87a57b6d86f43002d8139 Mon Sep 17 00:00:00 2001
+From: Bradford Zhang <zyc@zyc.name>
```
Comment (1715173329) [inline lang/ruby/patches/001-fix-build-with-libressl-3.5.patch:2]: what? this is not good. please keep original authorship.

File: lang/ruby/patches/001-fix-build-with-libressl-3.5.patch [hunk 2]
```diff
@@ -9,7 +9,7 @@ Subject: [PATCH] Fix build with LibreSSL 3.5
 
 --- a/ext/openssl/ossl_pkey.c
 +++ b/ext/openssl/ossl_pkey.c
-@@ -670,7 +670,7 @@ ossl_pkey_export_traditional(int argc, V
+@@ -710,7 +710,7 @@ ossl_pkey_export_traditional(int argc, VALUE *argv, VALUE self, int to_der)
```
Comment (1715173329) [inline lang/ruby/patches/001-fix-build-with-libressl-3.5.patch:12]: run `make package/ruby/refresh V=s` to refresh patches.

**Review by 1715173329 — CHANGES_REQUESTED** [review]

File: lang/ruby/Makefile [hunk 3]
```diff
@@ -11,15 +11,15 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=ruby
-PKG_VERSION:=3.1.2
+PKG_VERSION:=3.1.3
 PKG_RELEASE:=$(AUTORELEASE)
```
[the remark this answers is not in the document]
Comment (bradfordzhang) [reply lang/ruby/Makefile:15]: Thanks but I give up

**Comment by luizluca** [comment]
Thanks @bradfordzhang . I'll take it from here (#20064). But, please, try it again next time. Minor version bumps are normally straightforward.  

The $(AUTORELEASE) change hit you mid air. It is not directly related to ruby (or any other package) but it was recently deprecated by openwrt team. Any changes that touches a package that uses it should replaced that with a fixed number,  manually incremented on each package change.  As we are changing the version number as well, the release number rewinds back to 1.

And the `make package/ruby/refresh V=s` would fix the patches automatically for you. It looks like you did it manually because the line was not truncated. And we don't change author when a patch is refreshed. If you needed to modify it, you can explicitly add your info after the original commit message (before the ---).

Your commit message had a strange content: "Add a comment about confusing code". Version bumps do not need to be very verbose but it is nice to add CVE numbers when they are fixed.

**Comment by bradfordzhang** [comment]
> Thanks @bradfordzhang . I'll take it from here (#20064). But, please, try it again next time. Minor version bumps are normally straightforward.
> 
> The $(AUTORELEASE) change hit you mid air. It is not directly related to ruby (or any other package) but it was recently deprecated by openwrt team. Any changes that touches a package that uses it should replaced that with a fixed number, manually incremented on each package change. As we are changing the version number as well, the release number rewinds back to 1.
> 
> And the `make package/ruby/refresh V=s` would fix the patches automatically for you. It looks like you did it manually because the line was not truncated. And we don't change author when a patch is refreshed. If you needed to modify it, you can explicitly add your info after the original commit message (before the ---).
> 
> Your commit message had a strange content: "Add a comment about confusing code". Version bumps do not need to be very verbose but it is nice to add CVE numbers when they are fixed.

Thanks a lot. I'll do it better next time. :-)

**Pull request closed without merging**